autopilot/contractor.go

```diff
@@ -495,119 +508,84 @@ func (c *contractor) runContractRenewals(cfg api.AutopilotConfig, blockHeight, c
+		"run contracts refreshes",
+		"torefresh", len(toRefresh),
 		"budget", budget,
 	)
 	defer func() {
 		c.logger.Debugw(
-			"contract formations completed",
-			"formed", len(formed),
+			"contracts refreshes completed",
+			"refreshed", len(refreshed),
 			"budget", budget,
 		)
 	}()
 
-	// create a map of used hosts
-	used := make(map[types.PublicKey]bool)
-	for _, contract := range active {
-		used[contract.HostKey()] = true
-	}
-
-	// fetch recommended txn fee
-	fee, err := c.ap.bus.RecommendedFee()
-	if err != nil {
-		return nil, err
-	}
-
-	// fetch candidate hosts
-	wanted := int(addLeeway(missing, leewayPctCandidateHosts))
-	candidates, err := c.candidateHosts(cfg, used, wanted)
-	if err != nil {
-		return nil, err
-	}
-
-	// calculate min/max contract funds
-	minInitialContractFunds, maxInitialContractFunds := initialContractFundingMinMax(cfg)
-
-	for h := 0; missing > 0 && h < len(candidates); h++ {
-		host := candidates[h]
+	for _, ci := range toRefresh {
+		// TODO: keep track of consecutive failures and break at some point
 
 		// break if the contractor was stopped
 		if c.isStopped() {
 			break
 		}
 
-		// fetch host settings
-		scan, err := c.ap.worker.RHPScan(host.PublicKey, host.NetAddress, 0)
+		// convenience variables
+		contract := ci.contract
+		settings := ci.settings
+		fcid := contract.ID
+		rev := contract.Revision
+		hk := contract.HostKey()
+
+		// calculate the renter funds
+		renterFunds, err := c.refreshFundingEstimate(cfg, ci)
 		if err != nil {
-			c.logger.Debugw(err.Error(), "hk", host.PublicKey)
+			c.logger.Errorw(fmt.Sprintf("could not get refresh funding estimate, err: %v", err), "hk", hk, "fcid", fcid)
 			continue
 		}
-		settings := scan.Settings
 
 		// check our budget
-		txnFee := fee.Mul64(estimatedFileContractTransactionSetSize)
-		renterFunds := c.initialContractFunding(settings, txnFee, minInitialContractFunds, maxInitialContractFunds)
 		if budget.Cmp(renterFunds) < 0 {
-			c.logger.Debugw(
-				"insufficient budget",
-				"budget", budget,
-				"needed", renterFunds,
-				"renewal", false,
-			)
+			c.logger.Debugw("insufficient budget", "budget", budget, "needed", renterFunds)
 			break
 		}
 
 		// calculate the host collateral
-		hostCollateral, err := calculateHostCollateral(cfg, settings, renterFunds, txnFee)
-		if err != nil {
-			// TODO: keep track of consecutive failures and break at some point
-			c.logger.Errorw(
-				fmt.Sprintf("failed contract formation, err : %v", err),
-				"hk", host.PublicKey,
-			)
+		hostCollateral := renewContractCollateral(settings, renterFunds, contract.FileSize(), 0)
+
+		// check if the added collateral is below the threshold
+		_, hostMissedPayout, _ := rhpv2.CalculatePayouts(rev.FileContract, hostCollateral, settings, contract.EndHeight())
+		if isBelowCollateralThreshold(cfg, settings, hostMissedPayout) {
+			c.logger.Errorw(fmt.Sprintf("renew failed, added host collateral (%v) is below threshold", hostMissedPayout), "hk", hk, "fcid", fcid)
```

Comment:
"refresh failed, refreshed contract's collateral is below the collateral threshold"